src/linear_algebra/finsupp.lean

```diff
@@ -632,3 +633,19 @@ lemma mem_span_finset {s : finset M} {x : M} :
     (show x ∈ span R (id '' (↑s : set M)), by rwa set.image_id) in
   ⟨v, hvx ▸ (finsupp.total_apply_of_mem_supported _ hvs).symm⟩,
 λ ⟨f, hf⟩, hf ▸ sum_mem _ (λ i hi, smul_mem _ _ $ subset_span hi)⟩
+
+universe u
+
+/-- An element `m ∈ M` is contained in the `R`-submodule spanned by a set `s ⊆ M`, if and only if
+`m` can be written as a finite `R`-linear combination of elements of `s`.
+The implementation uses `finsupp.sum`. -/
+-- The Type `M` has an explicit universe, since otherwise it gets assigned `Type (max u_2 u_3)`.
+lemma mem_span_set {m : M} {s : set M} :
+  m ∈ submodule.span R s ↔
+  ∃ c : M →₀ R, (c.support : set M) ⊆ s ∧ (c.sum (λ i, (smul_add_hom R M).flip i)) = m :=
```

Comment:
Is there any reason to write this with `smul_add_hom` any more? This would feel more natural as
```suggestion
  ∃ c : M →₀ R, (c.support : set M) ⊆ s ∧ c.sum (λ mi r, r • mi) = m :=
```